src/cryptsetup/cryptsetup-generator.c

```diff
@@ -46,10 +45,36 @@ STATIC_DESTRUCTOR_REGISTER(arg_disks, hashmap_freep);
 STATIC_DESTRUCTOR_REGISTER(arg_default_options, freep);
 STATIC_DESTRUCTOR_REGISTER(arg_default_keyfile, freep);
 
-static int generate_keydev_mount(const char *name, const char *keydev, char **unit, char **mount) {
-        _cleanup_free_ char *u = NULL, *what = NULL, *where = NULL, *name_escaped = NULL;
+static int split_keyspec(const char *keyspec, char **keyfile, char **keydev) {
+        char *c;
+
+        assert(keyspec);
+        assert(keyfile);
+        assert(keydev);
+
+        c = strrchr(keyspec, ':');
+        if (c) {
+                *keyfile = strndup(keyspec, c-keyspec);
+                *keydev = strdup(c + 1);
```

Comment:
please do not clobber return values on failure, see:

https://systemd.io/CODING_STYLE#code-organization-and-semantics

i.e. fill in return parameters on on success, leave them unmodified on failure. That's the logic we generally follow in our codebase, and we should do so here too